engine/resource/src/providers/provider_zip.cpp

```diff
@@ -129,21 +92,69 @@ static void CreateEntryMap(ZipProviderContext* archive)
 
         const char* name = dmZip::GetEntryName(zip);
 
-        char archivepath[dmResource::RESOURCE_PATH_MAX];
-        dmSnPrintf(archivepath, sizeof(archivepath), "%s%s", name[0] != '/' ? "/" : "", name);
-        dmhash_t name_hash = dmHashString64(archivepath);
+        char archive_path[dmResource::RESOURCE_PATH_MAX];
+        dmSnPrintf(archive_path, sizeof(archive_path), "%s%s", name[0] != '/' ? "/" : "", name);
+        dmhash_t archive_path_hash = dmHashBufferNoReverse64(archive_path, strlen(archive_path));
 
         EntryInfo info;
         info.m_ManifestEntry = 0;
         dmZip::GetEntrySize(zip, &info.m_Size);
         dmZip::GetEntryIndex(zip, &info.m_EntryIndex);
 
-        DM_RESOURCE_DBG_LOG(3, "Added extra entry: %s %llx (%u bytes)\n", archivepath, name_hash, info.m_Size);
-
         dmZip::CloseEntry(zip);
 
-        archive->m_EntryMap.Put(name_hash, info);
+        temp_archive_map.Put(archive_path_hash, info);
     }
+
+    dmLiveUpdateDDF::HashAlgorithm algorithm = archive->m_Manifest->m_DDFData->m_Header.m_ResourceHashAlgorithm;
+    uint32_t hash_len = dmResource::HashLength(algorithm);
+
+    uint32_t manifest_entry_count = archive->m_Manifest->m_DDFData->m_Resources.m_Count;
+
+    dmHashTable64<EntryInfo>* entry_map = &archive->m_EntryMap;
+    uint32_t map_size = archive_entry_count + manifest_entry_count;
+    entry_map->SetCapacity(dmMath::Max(1U, (map_size*2)/3), map_size);
+
+    // Iterate over manifest entries. Different entries may refer to the same file in archive.
+    for (uint32_t i = 0; i < manifest_entry_count; ++i)
+    {
+        dmLiveUpdateDDF::ResourceEntry* entry = &archive->m_Manifest->m_DDFData->m_Resources.m_Data[i];
+        char name_buffer[dmResourceArchive::MAX_HASH*2+1];
+        dmResource::BytesToHexString(entry->m_Hash.m_Data.m_Data, hash_len, name_buffer, sizeof(name_buffer));
+        char archive_path_buffer[dmResourceArchive::MAX_HASH*2+1];
+        dmSnPrintf(archive_path_buffer, dmResourceArchive::MAX_HASH*2, "%s%s", name_buffer[0] != '/' ? "/" : "", name_buffer);
+        archive_path_buffer[dmResourceArchive::MAX_HASH*2] = 0;
+        dmhash_t archive_path_hash = dmHashBufferNoReverse64(archive_path_buffer, strlen(archive_path_buffer));
+        EntryInfo* info = temp_archive_map.Get(archive_path_hash);
+        if (!info)
+        {
+            // There is no such file in this archive
+            continue;
+        }
+        info->m_ManifestEntry = entry;
+        EntryInfo manifest_info;
+        manifest_info.m_ManifestEntry = entry;
+        manifest_info.m_Size = info->m_Size;
+        manifest_info.m_EntryIndex = info->m_EntryIndex;
```

Comment:
Create a copy of the `info` because one file may be referred by many manifest entries (I'm writing this comment and think that maybe it's even better if we use only one info for the same file? This map is read only anyway and will save a few kilobytes, not sure)